### PR TITLE
DNS integration

### PR DESCRIPTION
**Pull request by stuggi**
Depends-On: https://github.com/openstack-k8s-operators/infra-operator/pull/39
Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/297
Jira: [OSP-23053](https://issues.redhat.com//browse/OSP-23053)

### CODE REVIEW

**Comment by stuggi** [comment]
TODO, probably as follow up to configure dns on edpm nodes via ansible job network job? right now its only set on initial vm create

**Comment by stuggi** [comment]
could be tested using:
`OPENSTACK_REPO=https://github.com/stuggi/openstack-operator.git OPENSTACK_BRANCH=dns INFRA_REPO=https://github.com/stuggi/infra-operator.git INFRA_BRANCH=dns make openstack_deploy`

**Comment by stuggi** [comment]
~~~
cp /alabama/install_yamls/out/operator/openstack-operator/config/samples/core_v1beta1_openstackcontrolplane.yaml /alabama/install_yamls/out/openstack/openstack/cr
cp /alabama/install_yamls/out/operator/infra-operator/config/samples/network_v1beta1_dnsdata.yaml /alabama/install_yamls/out/openstack/openstack/cr # TODO (mschuppert): register edpm nodes in DNS can be removed after full IPAM integration
cp: cannot stat '/alabama/install_yamls/out/operator/infra-operator/config/samples/network_v1beta1_dnsdata.yaml': No such file or directory
~~~
its definitely there...
https://github.com/openstack-k8s-operators/infra-operator/blob/main/config/samples/network_v1beta1_dnsdata.yaml

**Comment by stuggi** [comment]
> ```
> cp /alabama/install_yamls/out/operator/openstack-operator/config/samples/core_v1beta1_openstackcontrolplane.yaml /alabama/install_yamls/out/openstack/openstack/cr
> cp /alabama/install_yamls/out/operator/infra-operator/config/samples/network_v1beta1_dnsdata.yaml /alabama/install_yamls/out/openstack/openstack/cr # TODO (mschuppert): register edpm nodes in DNS can be removed after full IPAM integration
> cp: cannot stat '/alabama/install_yamls/out/operator/infra-operator/config/samples/network_v1beta1_dnsdata.yaml': No such file or directory
> ```
> 
> its definitely there... https://github.com/openstack-k8s-operators/infra-operator/blob/main/config/samples/network_v1beta1_dnsdata.yaml

now got it in this case `NETWORK_ISOLATION=false` and the `dns_deploy_prep` is not running due to the condition in `openstack_prep`

File: Makefile [hunk 1]
```diff
@@ -472,6 +479,11 @@ edpm_deploy: input edpm_deploy_prep ## installs the dataplane instance using kus
 	$(eval $(call vars,$@,dataplane))
 	oc kustomize ${DEPLOY_DIR} | oc apply -f -
 
+.PHONY: edpm_register_dns
+edpm_register_dns: dns_deploy_prep ## register edpm nodes in dns as dnsdata
+	$(eval $(call vars,$@,infra))
+	oc apply -f ${DEPLOY_DIR}/network_v1beta1_dnsdata.yaml # TODO (mschuppert): register edpm nodes in DNS can be removed after full IPAM integration
```
Comment (abays) [inline Makefile:485]: Could the hardcode `ctlplane` values in this sample be a problem if the user changes the `DATAPLANE_COMPUTE_IP` or `DATAPLANE_COMPUTE_1_IP` variables?

Comment (stuggi) [reply Makefile:485]: yes, if someone changes the default IPs for the edpm nodes, they'd have to be registered manually via a dnsdata CR. running this target should only be a temporary until its integrated and we can remove it here.

**Comment by stuggi** [comment]
@slagle @rebtoor please have a look from edpm side. we can remove now static hosts entries with this.

**Comment by rebtoor** [comment]
lgtm for me, @slagle ?

**Comment by stuggi** [comment]
just for reference the doc https://github.com/openstack-k8s-operators/docs/blob/main/networking.md#dns

File: Makefile (hunk 1 above)
Comment (fao89) [reply Makefile:485]: @fultonj  the IPAM work was merged, so it might change

**Pull request merged**